ldap/servers/slapd/dyncerts.h

```diff
@@ -0,0 +1,158 @@
+/** BEGIN COPYRIGHT BLOCK
+ * Copyright (C) 2025 Red Hat, Inc.
+ * All rights reserved.
+ *
+ * License: GPL (version 3 or any later version).
+ * See LICENSE for details.
+ * END COPYRIGHT BLOCK **/
+
+#ifndef DYNCERT_H_
+#define DYNCERT_H_
+/*
+ * dyncerts.h - Dynamic Certificates Handler
+ */
+
+#define DYNCERTS_BENAME "DYNCERTS"
+#define DYNCERTS_BETYPE "DSE"
+
+#define CONFIG_DN1 "cn=config"
+#define CONFIG_DN2 "cn=encryption,cn=config"
+#define CONFIG_DN2_FILTER "(&(objectclass=nsEncryptionModule)(nsSSLActivation=on))"
+
+/* Indexes in config search set */
+#define CONFIG_ENTRY_IDX 	            0
+#define ENCRYPTION_CONFIG_ENTRY_IDX 	1
+#define FIRST_FAMILY_CONFIG_ENTRY_IDX 	2
+
+/* Conditionnaly add an attribute value */
+#define COND_STR(e, a, v, c) do { if (c) { slapi_entry_add_string((e), (a), (v)); } } while (false)
+
+/* The Dynamic certificate attribute names */
+#define DYCATTR_NICKNAME		 "cn"
+#define DYCATTR_PREFIX           "nsDynamicCertificate"
```

Comment:
As agreed in the design doc, let's use `ds` prefix instead of `ns`
```suggestion
#define DYCATTR_PREFIX           "dsDynamicCertificate"
```